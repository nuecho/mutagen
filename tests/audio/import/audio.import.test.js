const { readCsv } = require("../audio.test.helpers.js");

const INVALID_FILE_PATH = getResourcePath("audio/import/invalid.csv");
const MISSING_AUDIO_FILES_PATH = getResourcePath("audio/import/missing-audio-files.csv");
const INVALID_PERSONALITY_PATH = getResourcePath("audio/import/invalid-personality.csv");
const VALID_FILE_PATH = getResourcePath("audio/import/import.csv");
const MINIMALIST_FILE_PATH = getResourcePath("audio/import/import-minimalist.csv");

const OUTPUT_PREFIX = "/tmp/mutagen/test/import";

test(`[mutagen audio import --help] should print import usage`, () => {
  const { code, output } = mutagen(`--env gax audio import --help`);

  expect(output).toMatchSnapshot("import usage");
  expect(code).toBe(0);
});

test(`[mutagen audio import] should print usage when file to import is not specified`, () => {
  const { code, output } = mutagen(`--env gax audio import`);

  expect(output).toMatchSnapshot("import usage");
  expect(code).not.toBe(0);
});

test(`[mutagen audio import] should fail when file to import does not exist`, () => {
  const { code, output } = mutagen(`--env gax audio import /tmp/foo`);

  expect(output).toMatchSnapshot("no such file or directory");
  expect(code).not.toBe(0);
});

test(`[mutagen audio import] should fail when file to import is not a valid CSV`, () => {
  const { code, output } = mutagen(`--env gax audio import ${INVALID_FILE_PATH}`);

  expect(output).toMatchSnapshot("invalid csv");
  expect(code).not.toBe(0);
});

test(`[mutagen audio import] should fail when some audio files are missing`, () => {
  const { code, output } = mutagen(`--env gax audio import ${MISSING_AUDIO_FILES_PATH}`);

  expect(output).toMatchSnapshot("missing audio files");
  expect(code).not.toBe(0);
});

test(`[mutagen audio import] should fail when the csv file references invalid personality`, () => {
  const { code, output } = mutagen(`--env gax audio import ${INVALID_PERSONALITY_PATH}`);

  expect(output).toMatchSnapshot("missing personality");
  expect(code).not.toBe(0);
});

test(`[mutagen audio import] should import messages and audio files`, () => {
testImportWithAudios(VALID_FILE_PATH, "import_test");
});

test(`[mutagen audio import] message type and description should be optional`, () => {
  testImportWithAudios(MINIMALIST_FILE_PATH, "import_minimalist_test");
});

test(`[mutagen audio import] should validate duplicated messages`, () => {
  const { code, output } = mutagen(`--env gax audio import ${VALID_FILE_PATH}`);

  expect(output).toMatchSnapshot("duplicated messages");
  expect(code).not.toBe(0);
});

function testImportWithAudios(importFilePath, messagesKey) {
  const importResult = mutagen(`--env gax audio import ${importFilePath}`);

  expect(importResult.output).toMatchSnapshot("number of messages imported");
  expect(importResult.code).toBe(0);

  const outputDirectory = `${OUTPUT_PREFIX}/${messagesKey}`;
  const exportResult = mutagen(`--env gax audio export --with-audios ${outputDirectory}/output.csv`);
  expect(exportResult.code).toBe(0);

  const findResult = exec(`find ${outputDirectory} -type f -name *${messagesKey}* | sort`);
  expect(findResult.output).toMatchSnapshot("export of the imported files");

  const catResult = readCsv(`${outputDirectory}/output.csv`, messagesKey);
  expect(catResult).toMatchSnapshot("re-export of the csv file");
}