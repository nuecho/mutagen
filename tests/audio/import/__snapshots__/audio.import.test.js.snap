// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen audio import --help] should print import usage: import usage 1`] = `
Object {
  "stderr": "",
  "stdout": "Usage: import [-?] inputFile
Import audio files in ARM
      inputFile               Input audio description file.
  -?, -h, --help              Show this help message.
",
}
`;

exports[`[mutagen audio import] message type and description should be optional: export of the imported files 1`] = `
Object {
  "stderr": "",
  "stdout": "/tmp/mutagen/test/import/import_minimalist_test/10/import_minimalist_test message name 1.wav
/tmp/mutagen/test/import/import_minimalist_test/10/import_minimalist_test message name 2.wav
",
}
`;

exports[`[mutagen audio import] message type and description should be optional: number of messages imported 1`] = `
Object {
  "stderr": "",
  "stdout": "Imported 2 messages.
",
}
`;

exports[`[mutagen audio import] message type and description should be optional: re-export of the csv file 1`] = `
Object {
  "stderr": "",
  "stdout": "name,type,description,messageArId,tenantId,10,11
\\"import_minimalist_test message name 1\\",ANNOUNCEMENT,,<removed-for-testing-purposes>,1,\\"10/import_minimalist_test message name 1.wav\\",
\\"import_minimalist_test message name 2\\",ANNOUNCEMENT,,<removed-for-testing-purposes>,1,\\"10/import_minimalist_test message name 2.wav\\",
",
}
`;

exports[`[mutagen audio import] should fail when file to import does not exist: no such file or directory 1`] = `
Object {
  "stderr": "/tmp/foo (No such file or directory)
",
  "stdout": "",
}
`;

exports[`[mutagen audio import] should fail when file to import is not a valid CSV: invalid csv 1`] = `
Object {
  "stderr": "Invalid input file.
",
  "stdout": "",
}
`;

exports[`[mutagen audio import] should fail when some audio files are missing: missing audio files 1`] = `
Object {
  "stderr": "Audio import failed: the following audio files don't exist
/usr/src/app/tests/resources/audio/import/11/MISSING AUDIO FILE.wav
",
  "stdout": "",
}
`;

exports[`[mutagen audio import] should fail when the csv file references invalid personality: missing personality 1`] = `
Object {
  "stderr": "Audio import failed: the following personalities don't exist
12
",
  "stdout": "",
}
`;

exports[`[mutagen audio import] should import messages and audio files: export of the imported files 1`] = `
Object {
  "stderr": "",
  "stdout": "/tmp/mutagen/test/import/import_test/10/import_test message name 1.wav
/tmp/mutagen/test/import/import_test/10/import_test message name 2.wav
/tmp/mutagen/test/import/import_test/10/import_test music name 3.wav
/tmp/mutagen/test/import/import_test/11/import_test message name 1.wav
/tmp/mutagen/test/import/import_test/11/import_test music name 1.wav
/tmp/mutagen/test/import/import_test/11/import_test music name 3.wav
",
}
`;

exports[`[mutagen audio import] should import messages and audio files: number of messages imported 1`] = `
Object {
  "stderr": "",
  "stdout": "Imported 5 messages.
",
}
`;

exports[`[mutagen audio import] should import messages and audio files: re-export of the csv file 1`] = `
Object {
  "stderr": "",
  "stdout": "name,type,description,messageArId,tenantId,10,11
\\"import_test music name 1\\",MUSIC,\\"this is the description for music name 1\\",<removed-for-testing-purposes>,1,,\\"11/import_test music name 1.wav\\"
\\"import_test message name 1\\",ANNOUNCEMENT,\\"this is the description for message name 1\\",<removed-for-testing-purposes>,1,\\"10/import_test message name 1.wav\\",\\"11/import_test message name 1.wav\\"
\\"import_test music name 2\\",MUSIC,\\"this is the description for music name 2\\",<removed-for-testing-purposes>,1,,
\\"import_test message name 2\\",ANNOUNCEMENT,\\"this is the description for message name 2\\",<removed-for-testing-purposes>,1,\\"10/import_test message name 2.wav\\",
\\"import_test music name 3\\",MUSIC,\\"this is the description for music name 3\\",<removed-for-testing-purposes>,1,\\"10/import_test music name 3.wav\\",\\"11/import_test music name 3.wav\\"
",
}
`;

exports[`[mutagen audio import] should print usage when file to import is not specified: import usage 1`] = `
Object {
  "stderr": "",
  "stdout": "Missing required parameter: inputFile
Usage: import [-?] inputFile
Import audio files in ARM
      inputFile               Input audio description file.
  -?, -h, --help              Show this help message.
",
}
`;

exports[`[mutagen audio import] should validate duplicated messages: duplicated messages 1`] = `
Object {
  "stderr": "Audio import failed: the following messages' names already exist on the gax server
import_test music name 1
import_test message name 1
import_test music name 2
import_test message name 2
import_test music name 3
",
  "stdout": "",
}
`;
